src/main/java/core/basesyntax/strategy/BalanceHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class BalanceHandler implements OperationHandler {
+    private static final String BALANCE = "b";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(BALANCE)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
+        return listOfFruits;
```

Comment:
Returning 'listOfFruits' after modifying it is not a good practice as it can lead to side effects. Consider creating a new list that will store the updated quantities and return that list instead.